### PR TITLE
feature(swap): price impact risk

### PR DESCRIPTION
**Pull request by SorinC6**
task: https://emurgo.atlassian.net/browse/YOMO-1039

### CODE REVIEW

File: apps/wallet-mobile/src/components/Icon/Icon.stories.tsx [hunk 1]
```diff
@@ -242,6 +242,8 @@ storiesOf('Icon', module).add('Gallery', () => {
         <Item icon={<Icon.Exchange />} title="Exchange" />
 
         <Item icon={<Icon.Close />} title="Close" />
+
+        <Item icon={<Icon.Warning />} title="Close" />
```
Comment (stackchain) [inline apps/wallet-mobile/src/components/Icon/Icon.stories.tsx]:
```suggestion
        <Item icon={<Icon.Warning />} title="Warning" />
```

File: apps/wallet-mobile/src/features/Swap/common/PriceImpact/PriceImpact.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+        <Text style={{paddingRight: 10}}>{Math.ceil(priceImpact * 100) / 100}%</Text>
+
+        <Text> {`(${actualPrice} ${pair})`}</Text>
+      </Text>
+    </View>
+  )
+}
+
+export default PriceImpact
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Swap/common/PriceImpact/PriceImpact.tsx]:
```suggestion
```

File: apps/wallet-mobile/src/features/Swap/common/PriceImpact/PriceImpactWarning.tsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+import {useTheme} from '../../../../theme'
+import {Theme} from '../../../../theme/types'
+import {useStrings} from '../strings'
+
+export interface Props {
+  onSubmit: () => void
+  isLoading: boolean
+}
+
+export const PriceImpactWraning = ({onSubmit, isLoading}: Props) => {
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Swap/common/PriceImpact/PriceImpactWarning.tsx]:
```suggestion
export const PriceImpactWarning = ({onSubmit, isLoading}: Props) => {
```

File: apps/wallet-mobile/src/features/Swap/common/helpers.ts [hunk 4]
```diff
@@ -95,3 +96,26 @@ export const sortTokensByName = (a: Balance.TokenInfo, b: Balance.TokenInfo, wal
...
+    warning: theme.color.yellow[500],
+    negative: theme.color.magenta[500],
+  }
+}
+export const priceImpactBannerColorObject = (theme: Theme) => {
+  return {
+    warning: theme.color.yellow[100],
+    negative: theme.color.magenta[100],
+  }
+}
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Swap/common/helpers.ts]: export it out to a hook `usePrice...` 
```suggestion
export const usePriceImpactStatusTheme = (status: SwapPriceImpactStatus) {
  const theme = useTheme()
  
  if (status === 'negative') {
    return {
      text: theme.color.magenta[500],
      background: theme.color.magenta[100]
    }
  } else if (status === 'warning') {
    return {
      text: theme.color.yellow[500],
      background: theme.color.yellow[100]
    }
  }
  
  return {
    // TODO: fix
    text: theme.color.yellow[500],
    background: theme.color.yellow[100]
  }
}
```

File: apps/wallet-mobile/src/features/Swap/common/helpers.ts [hunk 5]
```diff
@@ -95,3 +96,26 @@ export const sortTokensByName = (a: Balance.TokenInfo, b: Balance.TokenInfo, wal
...
+
+export const calculatePriceImpactRisk = (priceImpact: number) => {
+  if (priceImpact >= 1 && priceImpact <= 10) {
+    return 'warning'
+  }
+  if (priceImpact > 10) {
+    return 'negative'
+  }
+  return 'positive'
+}
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Swap/common/helpers.ts]:
```suggestion
export const getPriceImpactStatus = (priceImpact: number) => {
  if (price < 1) return 'positive'
  if (price > 10) return 'negative'
  return 'warning'
}
```

File: apps/wallet-mobile/ios/yoroi.xcodeproj/project.pbxproj
Comment (stackchain) [inline apps/wallet-mobile/ios/yoroi.xcodeproj/project.pbxproj:1]: revert

File: apps/wallet-mobile/src/TxHistory/TxHistoryList/RampOnOffBanner/strings.ts
Comment (stackchain) [inline apps/wallet-mobile/src/TxHistory/TxHistoryList/RampOnOffBanner/strings.ts:1]: fixed in my branch can be reverted

File: apps/wallet-mobile/src/components/AmountItem/AmountItem.tsx [hunk 6]
```diff
@@ -18,20 +19,30 @@ export type AmountItemProps = {
...
 
   const isPrimary = tokenInfo.id === wallet.primaryTokenInfo.id
   const name = tokenInfo.ticker ?? tokenInfo.name
   const nameLabel = isEmptyString(name) ? '-' : name
   const detail = isPrimary ? tokenInfo.description : tokenInfo.fingerprint
 
   const formattedQuantity = Quantities.format(quantity, tokenInfo.decimals ?? 0)
-
   const showSwapDetails = !isPrimary && variant === 'swap'
+  const priceImpactColor = priceImpactColorObject(theme)
```
Comment (stackchain) [inline apps/wallet-mobile/src/components/AmountItem/AmountItem.tsx]: ⬆️

File: apps/wallet-mobile/src/features/Swap/common/strings.ts [hunk 7]
```diff
@@ -61,6 +62,10 @@ export const useStrings = () => {
     found: intl.formatMessage(messages.found),
     youHave: intl.formatMessage(messages.youHave),
     price: intl.formatMessage(messages.price),
+    priceimpact: intl.formatMessage(messages.priceimpact),
+    priceImpactOver10: intl.formatMessage(messages.priceImpactOver10),
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Swap/common/strings.ts]: priceImpactNegative? | Warning

File: apps/wallet-mobile/src/features/Swap/useCases/ConfirmTxScreen/TransactionSummary.tsx [hunk 8]
```diff
@@ -7,21 +7,36 @@ import {Icon, Spacer, Text, useModal} from '../../../../components'
...
-  const {amounts, selectedPoolCalculation: calculation} = orderData
+  const {theme} = useTheme()
+  const {
+    limitPrice: {displayValue: limitDisplayValue},
+  } = useSwapForm()
+  const {amounts, selectedPoolCalculation: calculation, calculations} = orderData
   const {openModal} = useModal()
 
+  const priceImpact = calculations[0]?.prices.priceImpact
+  const actualPrice = calculations[0]?.prices.actualPrice
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Swap/useCases/ConfirmTxScreen/TransactionSummary.tsx]: calculation

File: apps/wallet-mobile/src/features/Swap/useCases/ConfirmTxScreen/TransactionSummary.tsx [hunk 9]
```diff
@@ -7,21 +7,36 @@ import {Icon, Spacer, Text, useModal} from '../../../../components'
...
+
+  const priceImpactColor = priceImpactColorObject(theme)
+  const priceImpactBannerColorMap = priceImpactBannerColorObject(theme)
+
+  const priceImpactRisk = calculatePriceImpactRisk(Number(priceImpact))
+  const warningColorHex = priceImpactColor[priceImpactRisk]
+
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Swap/useCases/ConfirmTxScreen/TransactionSummary.tsx]: move below - > `if (!calculation)`

File: apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/CreateOrder/CreateOrder.tsx [hunk 10]
```diff
@@ -45,6 +47,7 @@ export const CreateOrder = () => {
   const {track} = useMetrics()
   const {openModal} = useModal()
   const {height: deviceHeight} = useWindowDimensions()
+  const priceImpactRisk = calculatePriceImpactRisk(Number(orderData.calculations[0]?.prices?.priceImpact))
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/CreateOrder/CreateOrder.tsx]: selectedPoolCalculation -> get...

**Review by stackchain — CHANGES_REQUESTED** [review]

File: apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/CreateOrder/CreateOrder.tsx [hunk 11]
```diff
@@ -211,6 +214,12 @@ export const CreateOrder = () => {
 
   const handleOnSwap = () => {
     if (orderData.selectedPoolCalculation === undefined) return
+
+    if (priceImpactRisk === 'negative') {
+      openModal(strings.warning, <PriceImpactWraning onSubmit={createUnsignedSwapTx} isLoading={isLoading} />, 400)
```
Comment (michalszorad) [inline apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/CreateOrder/CreateOrder.tsx]: Note the `isLoading` may not be passed down to `PriceImpactWraning ` when the state changes, as opening the modal is a one-time event.

File: apps/wallet-mobile/src/features/Swap/useCases/ConfirmTxScreen/TransactionSummary.tsx [hunk 12]
```diff
@@ -55,18 +73,61 @@ export const TransactionSummary = () => {
...
-      value: formattedFeeText,
-      info: strings.swapFees,
+      label: strings.marketPrice,
+      value: `${limitDisplayValue} ${tokenToSellName}/${tokenToBuyName}`,
+      info: strings.marketPriceInfo,
+    },
+    {
+      label: strings.priceimpact,
+      value:
+        priceImpactRisk === 'positive' ? (
```
Comment (michalszorad) [inline apps/wallet-mobile/src/features/Swap/useCases/ConfirmTxScreen/TransactionSummary.tsx]: Is it worth extracting the value into a component to improve readability?

File: apps/wallet-mobile/src/features/Swap/common/helpers.ts [hunk 13]
```diff
@@ -95,3 +96,26 @@ export const sortTokensByName = (a: Balance.TokenInfo, b: Balance.TokenInfo, wal
 
   return nameA.localeCompare(nameB, undefined, {sensitivity: 'base'})
 }
+
+export const calculatePriceImpactRisk = (priceImpact: number) => {
+  if (priceImpact >= 1 && priceImpact <= 10) {
+    return 'warning'
```
Comment (michalszorad) [inline apps/wallet-mobile/src/features/Swap/common/helpers.ts]: Is there a type for these strings? Can't find it. Something like `type PriceImpact = 'warning' | 'negative' | 'positive'`

File: apps/wallet-mobile/src/features/Swap/common/helpers.ts [hunk 14]
```diff
@@ -95,3 +97,30 @@ export const sortTokensByName = (a: Balance.TokenInfo, b: Balance.TokenInfo, wal
 
   return nameA.localeCompare(nameB, undefined, {sensitivity: 'base'})
 }
+
+export const getPriceImpactStatus = (priceImpact: number) => {
+  if (priceImpact < 1) return 'positive'
+  if (priceImpact > 10) return 'negative'
+  return 'warning'
+}
```
Comment (SorinC6) [inline apps/wallet-mobile/src/features/Swap/common/helpers.ts:105]: @stackchain should we let this helper here or will be better to move it in the swap package?

Comment (stackchain) [reply apps/wallet-mobile/src/features/Swap/common/helpers.ts:105]: I'd say to leave it here, since it's a completely UI/client decision.

**Review by michalszorad — APPROVED** [review]

File: apps/wallet-mobile/src/components/AmountItem/AmountItem.tsx [hunk 15]
```diff
@@ -18,9 +20,18 @@ export type AmountItemProps = {
   status?: string
   inWallet?: boolean
   variant?: 'swap'
+  priceImpactRisk?: SwapPriceImpactStatus
```
Comment (stackchain) [inline apps/wallet-mobile/src/components/AmountItem/AmountItem.tsx]: hey @SorinC6 I think your denomination is more precise let's rename `SwapPriceImpactStatus` to `SwapPriceImpactRisk` -> `'none' | 'high' | 'moderate'`

File: apps/wallet-mobile/src/components/AmountItem/AmountItem.tsx [hunk 16]
```diff
@@ -30,8 +41,8 @@ export const AmountItem = ({isPrivacyOff, wallet, style, amount, inWallet, varia
   const detail = isPrimary ? tokenInfo.description : tokenInfo.fingerprint
 
   const formattedQuantity = Quantities.format(quantity, tokenInfo.decimals ?? 0)
-
   const showSwapDetails = !isPrimary && variant === 'swap'
+  const priceImpactColor = usePriceImpactStatusTheme(priceImpactRisk ?? 'positive')
```
Comment (stackchain) [inline apps/wallet-mobile/src/components/AmountItem/AmountItem.tsx]:
```suggestion
  const priceImpactRiskTheme = usePriceImpactRiskTheme(priceImpactRisk)
```

File: apps/wallet-mobile/src/features/Swap/common/PriceImpact/PriceImpact.tsx [hunk 17]
```diff
@@ -0,0 +1,53 @@
...
+type PriceImactProps = {
+  priceImpact: number
+  actualPrice: number
+  pair: string
+}
+
+export const PriceImpact = ({priceImpact, actualPrice, pair}: PriceImactProps) => {
+  const strings = useStrings()
+  const priceImpactRisk = getPriceImpactStatus(priceImpact)
+  const priceImpactColor = usePriceImpactStatusTheme(priceImpactRisk ?? 'positive')
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Swap/common/PriceImpact/PriceImpact.tsx]:
```suggestion
  const priceImpactRiskTheme = usePriceImpactRiskTheme(priceImpactRisk)
```

File: apps/wallet-mobile/src/features/Swap/common/PriceImpact/PriceImpactWarning.tsx [hunk 18]
```diff
@@ -0,0 +1,77 @@
+import React from 'react'
+import {StyleSheet, Text, View} from 'react-native'
+
+import {Button, Spacer, useModal} from '../../../../components'
+import {useTheme} from '../../../../theme'
+import {Theme} from '../../../../theme/types'
+import {useStrings} from '../strings'
+
+export interface Props {
+  onSubmit: () => void
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Swap/common/PriceImpact/PriceImpactWarning.tsx]:
```suggestion
  onContinue: () => void
```

File: apps/wallet-mobile/src/features/Swap/common/PriceImpact/PriceImpactWarning.tsx [hunk 19]
```diff
@@ -0,0 +1,77 @@
...
+}
+
+export const PriceImpactWarning = ({onSubmit}: Props) => {
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Swap/common/PriceImpact/PriceImpactWarning.tsx]:
```suggestion
export const PriceImpactWarning = ({onContinue}: Props) => {
```

File: apps/wallet-mobile/src/features/Swap/common/helpers.ts [hunk 20]
```diff
@@ -95,3 +97,30 @@ export const sortTokensByName = (a: Balance.TokenInfo, b: Balance.TokenInfo, wal
 
   return nameA.localeCompare(nameB, undefined, {sensitivity: 'base'})
 }
+
+export const getPriceImpactStatus = (priceImpact: number) => {
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Swap/common/helpers.ts]:
```suggestion
export const getPriceImpactRisk = (priceImpact: number) => {
```

File: apps/wallet-mobile/src/features/Swap/common/helpers.ts (hunk 14 above)
Comment (stackchain) [inline apps/wallet-mobile/src/features/Swap/common/helpers.ts]:
```suggestion
  if (priceImpact < 1) return 'none'
  if (priceImpact > 10) return 'high'
  return 'moderate'
}
```

File: apps/wallet-mobile/src/features/Swap/common/helpers.ts [hunk 21]
```diff
@@ -95,3 +97,30 @@ export const sortTokensByName = (a: Balance.TokenInfo, b: Balance.TokenInfo, wal
...
+      background: theme.color.magenta[100],
+    }
+  } else if (status === 'warning') {
+    return {
+      text: theme.color.yellow[500],
+      background: theme.color.yellow[100],
+    }
+  }
+
+  return {
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Swap/common/helpers.ts:122]: this should return the default colours for `none` - black (even if its not on the screen for now) when is `none`

File: apps/wallet-mobile/src/features/Swap/common/helpers.ts [hunk 22]
```diff
@@ -95,3 +97,30 @@ export const sortTokensByName = (a: Balance.TokenInfo, b: Balance.TokenInfo, wal
...
+export const getPriceImpactStatus = (priceImpact: number) => {
+  if (priceImpact < 1) return 'positive'
+  if (priceImpact > 10) return 'negative'
+  return 'warning'
+}
+
+export const usePriceImpactStatusTheme = (status: SwapPriceImpactStatus) => {
+  const {theme} = useTheme()
+
+  if (status === 'negative') {
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Swap/common/helpers.ts]:
```suggestion
  if (status === 'high') {
```

File: apps/wallet-mobile/src/features/Swap/common/helpers.ts [hunk 23]
```diff
@@ -95,3 +97,30 @@ export const sortTokensByName = (a: Balance.TokenInfo, b: Balance.TokenInfo, wal
...
   return nameA.localeCompare(nameB, undefined, {sensitivity: 'base'})
 }
+
+export const getPriceImpactStatus = (priceImpact: number) => {
+  if (priceImpact < 1) return 'positive'
+  if (priceImpact > 10) return 'negative'
+  return 'warning'
+}
+
+export const usePriceImpactStatusTheme = (status: SwapPriceImpactStatus) => {
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Swap/common/helpers.ts]:
```suggestion
export const usePriceImpactStatusTheme = (risk: SwapPriceImpactRisk) => {
```

File: apps/wallet-mobile/src/features/Swap/common/helpers.ts [hunk 24]
```diff
@@ -95,3 +97,30 @@ export const sortTokensByName = (a: Balance.TokenInfo, b: Balance.TokenInfo, wal
...
+  const {theme} = useTheme()
+
+  if (status === 'negative') {
+    return {
+      text: theme.color.magenta[500],
+      background: theme.color.magenta[100],
+    }
+  } else if (status === 'warning') {
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Swap/common/helpers.ts]:
```suggestion
  } else if (status === 'moderate') {
```

File: apps/wallet-mobile/src/features/Swap/common/strings.ts [hunk 25]
```diff
@@ -61,6 +62,10 @@ export const useStrings = () => {
     found: intl.formatMessage(messages.found),
     youHave: intl.formatMessage(messages.youHave),
     price: intl.formatMessage(messages.price),
+    priceimpact: intl.formatMessage(messages.priceimpact),
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Swap/common/strings.ts]:
```suggestion
    priceImpact: intl.formatMessage(messages.priceImpact),
```

File: apps/wallet-mobile/src/features/Swap/common/strings.ts [hunk 26]
```diff
@@ -61,6 +62,10 @@ export const useStrings = () => {
...
+    priceImpactNegative: intl.formatMessage(messages.priceImpactNegative),
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Swap/common/strings.ts]:
```suggestion
    priceImpactRiskHigh: intl.formatMessage(messages.priceImpactRiskHigh),
```

File: apps/wallet-mobile/src/features/Swap/common/strings.ts [hunk 27]
```diff
@@ -601,4 +617,26 @@ export const messages = defineMessages({
     id: 'swap.listOrders.emptyCompletedOrders',
     defaultMessage: '!!!No orders completed yet',
   },
+  priceimpact: {
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Swap/common/strings.ts]:
```suggestion
  priceImpact: {
```

File: apps/wallet-mobile/src/features/Swap/common/strings.ts [hunk 28]
```diff
@@ -601,4 +617,26 @@ export const messages = defineMessages({
...
+    id: 'swap.swapScreen.priceImpact',
+    defaultMessage: '!!!Price Impact',
+  },
+
+  priceImpactNegative: {
+    id: 'swap.swapScreen.priceImpactNegative',
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Swap/common/strings.ts]:
```suggestion
  priceImpactRiskHigh: {
    id: 'swap.swapScreen.priceImpactRiskHigh',
```

File: apps/wallet-mobile/src/features/Swap/useCases/ConfirmTxScreen/TransactionSummary.tsx [hunk 29]
```diff
@@ -10,22 +10,33 @@ import {useSelectedWallet} from '../../../../SelectedWallet'
...
+    limitPrice: {displayValue: limitDisplayValue},
+  } = useSwapForm()
+  const {amounts, selectedPoolCalculation: calculation, calculations} = orderData
   const {openModal} = useModal()
 
   // should never happen
   if (!calculation) throw new Error('No selected pool calculation')
   const {pool, cost} = calculation
 
+  const priceImpact = calculations[0]?.prices.priceImpact
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Swap/useCases/ConfirmTxScreen/TransactionSummary.tsx]: calculation (has all the calculations), it should use on summary, calculation, which is the selected calculation.

Comment (stackchain) [inline apps/wallet-mobile/src/features/Swap/useCases/ConfirmTxScreen/TransactionSummary.tsx]:
```suggestion
  const priceImpact = calculation.pool.prices.priceImpact
```

File: apps/wallet-mobile/src/i18n/locales/en-US.json [hunk 30]
```diff
@@ -823,6 +824,10 @@
   "swap.swapScreen.notEnoughSupply": "Not enough supply in the pool",
   "swap.swapScreen.openOrders": "Open orders",
   "swap.swapScreen.ordersSwapTab": "Orders",
+  "swap.swapScreen.priceImpact": "Price Impact",
+  "swap.swapScreen.priceImpactInfo": "Price impact is a difference between the actual market price and your price due to trade size.",
+  "swap.swapScreen.priceImpactNegative": "Price impact over 10%",
```
Comment (stackchain) [inline apps/wallet-mobile/src/i18n/locales/en-US.json]:
```suggestion
  "swap.swapScreen.priceImpactRiskHigh": "Price impact over 10%",
```

File: apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/CreateOrder/CreateOrder.tsx [hunk 31]
```diff
@@ -211,6 +214,12 @@ export const CreateOrder = () => {
 
   const handleOnSwap = () => {
     if (orderData.selectedPoolCalculation === undefined) return
+
+    if (priceImpactRisk === 'negative') {
+      openModal(strings.warning, <PriceImpactWarning onSubmit={createUnsignedSwapTx} />, 400)
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/CreateOrder/CreateOrder.tsx]:
```suggestion
      openModal(strings.warning, <PriceImpactWarning onContinue={createUnsignedSwapTx} />, 400)
```

File: apps/wallet-mobile/src/features/Swap/common/PriceImpact/PriceImpact.tsx [hunk 32]
```diff
@@ -0,0 +1,53 @@
+import React from 'react'
+import {StyleSheet, Text, View} from 'react-native'
+
+import {Icon} from '../../../../components'
+import {getPriceImpactStatus, usePriceImpactStatusTheme} from '../helpers'
+import {useStrings} from '../strings'
+
+type PriceImactProps = {
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Swap/common/PriceImpact/PriceImpact.tsx]:
```suggestion
type PriceImpactProps = {
```

File: apps/wallet-mobile/src/features/Swap/common/PriceImpact/PriceImpact.tsx [hunk 33]
```diff
@@ -0,0 +1,53 @@
...
+  priceImpact: number
+  actualPrice: number
+  pair: string
+}
+
+export const PriceImpact = ({priceImpact, actualPrice, pair}: PriceImactProps) => {
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Swap/common/PriceImpact/PriceImpact.tsx]:
```suggestion
export const PriceImpact = ({priceImpact, actualPrice, pair}: PriceImpactProps) => {
```

File: apps/wallet-mobile/src/features/Swap/useCases/ConfirmTxScreen/TransactionSummary.tsx [hunk 34]
```diff
@@ -131,13 +188,31 @@ export const TransactionSummary = () => {
...
+              </View>
             </View>
           </View>
         )
       })}
 
-      <Spacer height={24} />
+      <Spacer height={12} />
+
+      {(priceImpactRisk === 'warning' || priceImpactRisk === 'negative') && (
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Swap/useCases/ConfirmTxScreen/TransactionSummary.tsx]:
```suggestion
      {priceImpactRisk !== 'none' && (
```

File: apps/wallet-mobile/src/features/Swap/common/PriceImpact/PriceImpact.tsx [hunk 35]
```diff
@@ -0,0 +1,53 @@
...
+      <View>
+        {priceImpactRisk === 'moderate' && <Icon.Info size={20} color={warningColorHex} />}
+
+        {priceImpactRisk === 'high' && <Icon.Warning size={20} color={warningColorHex} />}
+      </View>
+
+      <Text style={{color: warningColorHex}}>
+        <Text>{strings.priceImpact}</Text>
+
+        <Text> = </Text>
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Swap/common/PriceImpact/PriceImpact.tsx:35]: 👀

File: apps/wallet-mobile/src/features/Swap/useCases/ConfirmTxScreen/TransactionSummary.tsx [hunk 36]
```diff
@@ -10,22 +10,33 @@ import {useSelectedWallet} from '../../../../SelectedWallet'
...
+  } = useSwapForm()
   const {amounts, selectedPoolCalculation: calculation} = orderData
   const {openModal} = useModal()
 
   // should never happen
   if (!calculation) throw new Error('No selected pool calculation')
   const {pool, cost} = calculation
 
+  const priceImpact = calculation.prices.priceImpact
+  const actualPrice = calculation.prices.actualPrice
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Swap/useCases/ConfirmTxScreen/TransactionSummary.tsx]:
```suggestion
  const {pool, cost, prices} = calculation

  const priceImpact = prices.priceImpact
  const actualPrice = prices.actualPrice
```

File: apps/wallet-mobile/src/features/Swap/common/PriceImpact/PriceImpact.tsx (hunk 35 above)
Comment (SorinC6) [reply apps/wallet-mobile/src/features/Swap/common/PriceImpact/PriceImpact.tsx:35]: was not sure what to do with the equal sign 🤔 as I think we can let it hardcoded as it is

Comment (stackchain) [reply apps/wallet-mobile/src/features/Swap/common/PriceImpact/PriceImpact.tsx:35]: is it wrapping?

**Pull request merged**